benchmark_tests/test_tutorial_docs.py

```diff
@@ -0,0 +1,75 @@
+import re
+import shutil
+import pytest
+
+@pytest.fixture(scope="function")
+def teardown(request):
+    """
+    Delete the folder at the end of the test
+    """
+    folder_name = request.param
+    yield
+    shutil.rmtree(folder_name)
+
+def find_first_greater(my_list, value):
+    """
+    Find the first value greater than the given value
+
+    Args:
+        my_list (list): list of values
+        value (float): value to compare
+    """
+    for item in my_list:
+        if item > value:
+            return item
+    return None
+
+def read_tutorial(rst_file, name):
+    """
+    Read the code from the rst file
+
+    Args:
+        rst_file (str): path to the rst file
+        name (str): name of the tutorial
+    """
+
+    with open(rst_file, "r") as fi:
+        lines = fi.read().splitlines()
+
+    # find start line of tutorial and end of tutorial
+    idx_ini = [i for i, val in enumerate(lines) if name in val][0]
+    idx_end = [i for i, val in enumerate(lines) if "_tutorial" in val]
+    idx_end = find_first_greater(idx_end, idx_ini)
+
+    tutorial = lines[idx_ini: idx_end]
+
+    # find start of python code in tutorial
```

Comment:
can you write in a comment how you find the start of the python code?

```suggestion
    # find start of python code in tutorial, by checking for: '.. code-block:: python'
```